cmake/modules/hwm_v2.cmake

```diff
@@ -0,0 +1,101 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (c) 2023, Nordic Semiconductor ASA
+
+# The HWMv2 CMake module works together with the list_hardware.py script to
+# obtain all archs and SoC implementations defined in the Zephyr build system.
+#
+# The result from list_hwm.py is then used to generate Kconfig HWMv2 and CMake
+# HWMv2 files for the build system.
+#
+# The following files are generated in '<kconfig-binary-dir>/'
+# - Kconfig.defconfig: Contains references to SoC v2 defconfig files for Zephyr integration.
+# - Kconfig: Contains references to regular SoC v2 Kconfig files for Zephyr integration.
+# - Kconfig.soc: Contains references to generic SoC v2 Kconfig files.
+
+include_guard(GLOBAL)
+
+if(NOT HWMv2)
+  return()
+endif()
+
+# Internal helper function for creation of Kconfig files.
+function(kconfig_soc_gen file dirs)
+  set(kconfig_file ${KCONFIG_BINARY_DIR}/${file})
+  foreach(dir ${dirs})
+    file(APPEND ${kconfig_file} "osource \"${dir}/${file}\"\n")
+  endforeach()
+endfunction()
+
+# 'SOC_ROOT' and 'ARCH_ROOT' are prioritized lists of directories where their
+# implementations may be found. It always includes ${ZEPHYR_BASE}/[arch|soc]
+# at the lowest priority.
+list(APPEND SOC_ROOT ${ZEPHYR_BASE})
+list(APPEND ARCH_ROOT ${ZEPHYR_BASE})
+
+list(TRANSFORM ARCH_ROOT PREPEND "--arch-root=" OUTPUT_VARIABLE arch_root_args)
+list(TRANSFORM SOC_ROOT PREPEND "--soc-root=" OUTPUT_VARIABLE soc_root_args)
+
+execute_process(COMMAND ${PYTHON_EXECUTABLE} ${ZEPHYR_BASE}/scripts/list_hardware.py
+                ${arch_root_args} ${soc_root_args}
+                --archs --socs
+                --cmakeformat={TYPE}\;{NAME}\;{DIR}\;{HWM}
+                OUTPUT_VARIABLE ret_hwm
+                ERROR_VARIABLE err_hwm
+                RESULT_VARIABLE ret_val
+)
+if(ret_val)
+  message(FATAL_ERROR "Error processing HWMv2.\nError message: ${err_board}")
```

Comment:
"error listing hardware"? I find all the "hwmv2" terminology exposed to the user a bit obscure, remember many people will have no clue on what "hwmv2" means. Also, considering this terminology will disappear soon (hwmv2 will quickly replace v1), I'd avoid usage of "hwmv2" wording as much as possible.

Reply:
Thanks for the suggestion, and I completely agree that places like this we should avoid the term HVMv2 / v2.

Suggestion applied.